js/src/javascript/beautifier.js

```diff
@@ -1412,12 +1413,6 @@ Beautifier.prototype.handle_dot = function(current_token) {
       this._flags.last_token.text === ')' && this._options.break_chained_methods);
   }
 
-  // Only unindent chained method dot if this dot starts a new line.
-  // Otherwise the automatic extra indentation removal will handle the over indent
-  if (this._options.unindent_chained_methods && this._output.just_added_newline()) {
-    this.deindent();
-  }
-
```

Comment:
This change makes me a bit nervous - it effectively makes a local setting for dots into a broader setting that could have side effects in a number of places.  

I'm going to take all the other changes in this PR now though.  Thanks!

Reply:
@bitwiseman, sorry for no explanation about this. The change is not related only to the dot. Problem is really deeper. I've traced behavior and found a bunch of problems with deindention, not with dot only. There is wrong behavior with not closed block statements within an expression.

For example:
(**unindent_chained_methods** is ON)
```
getNum(() => {
    return 10;
}) -
5;
```
Expected output:
```
getNum(() => {
    return 10;
}) -
5;
```
Actual output:
```
getNum(() => {
        return 10;
    }) -
    5;
```

So my fix works for all cases, not only for dots. I can understand your nervous because this is pretty not obvious and complicated :)

Reply:
@HanabishiRecca 
Your explanation  doesn't include a dot, but the change is based on `unindent_chained_methods` ... So, that means that even though the example doesn't include chained methods the behavior will change depending on whether `unindent_chained_methods` is true or false... 

You arrived because of #1651  right?

Reply:
Yes. I can agree this is a bit weird. If we want to separate this behavior, this logic really will be wrong. Ok, I will try to find another solution.